lib/noticed/base.rb

```diff
@@ -71,22 +71,19 @@ def params
     def run_delivery(recipient, enqueue: true)
       delivery_methods = self.class.delivery_methods.dup
 
-      # Set recipient to instance var so it is available to Notification class
-      @recipient = recipient
-
       # Run database delivery inline first if it exists so other methods have access to the record
       if (index = delivery_methods.find_index { |m| m[:name] == :database })
         delivery_method = delivery_methods.delete_at(index)
-        @record = run_delivery_method(delivery_method, recipient: recipient, enqueue: false)
```

Comment:
Since we have an `attr_accessor :record` on line 13, assigning to the instance variable makes the record available in all methods later on. We don't need to pass it as an argument since it's an instance variable.

Reply:
@excid3 But I think this is not being used because we are over-writing the accessor again over [here](https://github.com/excid3/noticed/blob/66d47aa1cc59d4271ec13d525703f5a8ec4b9107/lib/noticed/delivery_methods/base.rb#L40)

"If the accessor's value is being set later, why set it here too?" is the question that led me to this refactor.

Reply:
Ok, I get it. Its not about the accessor, you'd prefer this to be an instance variable simply because we'd not have to pass `record` around as arguments to methods.

In that case, what do you think about following the same pattern for `recipient` too? The fact that `recipient` is being passed around as argument, while `record` is not is confusing. What do you think about having a similar approach for both - ie, either both remain as instance variables, or both are passed around as arguments? 🙂

Reply:
We do actually set the recipient instance variable too: https://github.com/excid3/noticed/blob/66d47aa1cc59d4271ec13d525703f5a8ec4b9107/lib/noticed/base.rb#L75

It's been a long time since I've looked at this code. I don't think we need to pass the recipient then either, or we could pass both. 

I know the main reason we have the attr_accessors for record and recipient on Noticed::Base is so that during a delivery, we can easily access those from the Notification object. As long as we still have that, I think it's fine.

Reply:
@excid3 personally, I'd prefer passing both as arguments. But happy to do this the other way if you'd prefer that, please let me know 🙂

Reply:
If there's anything else that needs fixing, let me know. Once this is merged, I can proceed to implement bulk deliveries 🙂